New docs website / Pagination (#1014)
hashicorp/design-system: website/docs/components/pagination/partials/guidelines/guidelines.md

## Usage

### When to use

- To break down large content into pages. Usually paired with tables, but works with other types of page content.

### When not to use

> - As a navigation control for a flow or to pair with a stepper, i.e. for a guide, tutorial, setup flow.

Reviewer: ```suggestion
- As a navigation control for a flow or to pair with a stepper, i.e. for a guide, tutorial, or setup flow.
```

- As a controller to switch between multiple views. Use [Tabs](/components/tabs/) instead.

> ### Numbered vs Compact

Reviewer: ```suggestion
### Numbered vs compact
```

Author: I'm hesitant about this one because we refer to to a variant name, so I feel like it's correct in this case, but I'm not sure. I can't find a similar example on another component. @MelSumner @heatherlarsen do you have an opinion on this?

Reviewer: Feel free to leave as is if you disagree. We could always update later if needed

Reviewer 2: Hmm, I suppose this is a question about whether or not we consider variant and property names as proper nouns. If we consider them proper nouns, then they should be capitalized. If not, then they should be lowercase. I'm leaning toward it being capitalized, but very curious about what others think.

**Reference**
"In title case, major words are capitalized, and most minor words are lowercase. In sentence case, most major and minor words are lowercase ([proper nouns](https://apastyle.apa.org/style-grammar-guidelines/capitalization/proper-nouns) are an exception in that they are always capitalized)." - [APA style](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)

> In most cases, the numbered pagination provides a greater user experience. It allows users to jump between pages and always return to the first page or go to the last page without navigating through the pages manually.

Reviewer: ```suggestion
In most cases, the numbered pagination provides a better user experience. It allows users to jump between pages and always return to the first page or go to the last page without navigating through the pages manually.
```

Author: I'm hesitant about this one because we refer to to a variant name, so I feel like it's correct in this case, but I'm not sure. I can't find a similar example on another component. @MelSumner @heatherlarsen do you have an opinion on this?

If your product only has cursor pagination, it won't be able to support the numbered variant. Only applications with offset pagination can implement the numbered pagination variant.

![Numbered pagination example](/assets/components/pagination/pagination-offset-example.png =50%x50%)
- Supported by offset (page-based) pagination.

![Compact pagination example](/assets/components/pagination/pagination-cursor-example.png =25%x25%)
- Supported by offset and cursor based pagination.

!!! Warning 

> We strongly recommend talking to your engineering team before using a specific variant. Learn more about [offset and cursor pagination] (https://ignaciochiazzo.medium.com/paginating-requests-in-apis-d4883d4c1c4c)

Reviewer 2: ```suggestion
We strongly suggest that you talk to your engineering team to see if offset or cursor pagination are right for your project. Learn more about [offset and cursor pagination] (https://ignaciochiazzo.medium.com/paginating-requests-in-apis-d4883d4c1c4c).
```

> !!!

Reviewer 2: It feels like we're trying to do too much in one section (i.e., cursor vs. offset and numbered vs. compact). Have you thought about how to provide more helpful information about cursor vs. offset pagination directly in the docs (vs. just linking out to an article)? Admittedly I had no idea what cursor pagination was until you told us about it in a design review.

Author: Thanks for reviewing. I'm addressing the other changes and will tackle this one in a separate commit as I need to think a bit about it.

### Spacing

- When using the pagination bar, the container should be flush on the left & right with the content.
- When using the pagination, the component should be center aligned with the content it relates to.
- Make sure there's enough distance and breathing room between the pagination and not related content (e.g. another section below it), so it's clear what content the pagination is paired with.

When pairing the pagination or pagination bar with tables, we recommend:

> - 12px or 16px between the pagination or pagination bar and the content it relates to.

Reviewer 2: Why is it 12px **or** 16px? Shouldn't we provide guidance that results in a more consistent experience here? If and when we turn this into a pattern the guidance would need to be consistent.

Author: I'll set it to 16px but I feel we may have to re-write the guidelines as products start to adopt it, mainly because the ideal spacing depends on the content, the table styling, and spacing scale of the product, which currently isn't consistent.

![Spacing for pagination paired with tables](/assets/components/pagination/pagination-spacing-tables.png)

> When pairing with not contained content (e.g. cards, lists), we recommend:

Reviewer: ```suggestion
When pairing with content which is not contained (e.g. cards, lists), we recommend:
```
